### PR TITLE
76 implement execute simple command from path

### LINKED ISSUE #76: Implement execute_simple_command from PATH

### LINKED ISSUE #77: Exec command flow
## 

``` c
cmd_path = argv[0];
if (get_env_entry(env, "PATH") && !has_slash(cmd_path))
{
        // returns full path of the first executable command (X_OK) from PATH
	cmd_path = search_path(cmd_path, env);
	if (cmd_path == NULL)
	{
		print_error(argv[0], CMD_NOT_FOUND);
		exit(127);
	}
}
// cmd_path exists
if (is_a_directory(cmd_path))
{
	print_error(cmd_path, strerror(EISDIR));
	exit(126);
}
if (execve(cmd_path, argv, (char **)convert_env_to_array(env)) == -1)
{
	print_error(cmd_path, strerror(errno));
	if (errno == ENOENT)
		exit(127);
	else
		exit(126);
}

```

### PR DESCRIPTION
**Pull request by rask24**
fix #76
fix #77

- **Implement exec_simple_command from PATH**
- **Update execute_simple_command**
- **Remove return_entire_path**
- **Remove xaccess**
- **Refactor convert_cmd_args_to_array**
- **Refactor print_error_exit**

## Release Notes

- **New Features**
  - Enhanced error handling capabilities with new functions for improved command execution.
  - Added a series of end-to-end tests for shell-like command execution, covering various commands and error scenarios.
  - Introduced a new function to convert command arguments from an abstract syntax tree into an array of strings.

- **Bug Fixes**
  - Improved command resolution logic and error handling during command execution.

- **Removals**
  - Removed outdated functionality related to path resolution and file access checks.
  - Deleted obsolete test cases that are no longer relevant.

- **Tests**
  - Introduced new tests for command execution and improved coverage of error handling scenarios. 
  - Added unit tests for the new command argument conversion function.

### CODE REVIEW

File: src/exec/execute_simple_command.c [hunk 1]
```diff
@@ -61,32 +45,82 @@
...
+	while (paths[i])
+	{
+		tmp = ft_xstrjoin(paths[i], "/");
+		joined_path = ft_xstrjoin(tmp, basename);
+		free(tmp);
+		if (access(joined_path, X_OK) == 0)
+		{
+			ft_free_strs(paths);
+			return (joined_path);
+		}
```
[the remark this answers is not in the document]
Comment (rask24) [reply src/exec/execute_simple_command.c:77]: Exactly. I've fixed it.

File: src/exec/execute_simple_command.c [hunk 2]
```diff
@@ -61,32 +45,82 @@
...
+		return (false);
+	return (S_ISDIR(st.st_mode));
+}
+
+static char	*search_for_command(char *basename, t_env_list *env_list)
+{
+	char	**paths;
+	char	*joined_path;
+	char	*tmp;
+	int		i;
```
[the remark this answers is not in the document]
Comment (rask24) [reply src/exec/execute_simple_command.c:62]: That's correct. However, `search_for_command` is only called in contexts where `lookup_value("PATH", env_list)` is NOT NULL. I've implemented it this way to maintain the coverage rate.

File: src/exec/execute_simple_command.c [hunk 3]
```diff
@@ -61,32 +45,82 @@
...
+			print_error_exit(argv[0], strerror(errno), EXIT_OTHER_ERR);
+	}
+}
+
 pid_t	execute_simple_command(t_ast *node, t_env_list *env_list,
 			int fd_in, int fd_out)
 {
-	pid_t	pid;
-	char	**argv;
+	pid_t		pid;
```
[the remark this answers is not in the document]
Comment (rask24) [reply src/exec/execute_simple_command.c:109]: The return value of the function is pid_t. So using EXIT_FAILURE could potentially cause unexpected behavior.
I think the essence of this issue lies in the design of the exec module. I'll address this problem in a separate issue.

**Review by alex216 — APPROVED** [review]
LGTM @rask24

**Pull request merged**